p2p/libp2p/ports.go

```diff
@@ -0,0 +1,90 @@
+package libp2p
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go/core/random"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+)
+
+type testPortHandler = func(int) error
+
+func getPort(port string, handler testPortHandler) (int, error) {
+	val, err := strconv.Atoi(port)
+	if err == nil {
+		if val < 0 {
+			return 0, fmt.Errorf("%w, %d does not represent a positive value for port", p2p.ErrInvalidPortValue, val)
+		}
+
+		return val, nil
+	}
+
+	ports := strings.Split(port, "-")
+	if len(ports) != 2 {
+		return 0, fmt.Errorf("%w, provided port string `%s` is not in the correct format, expected `start-end`", p2p.ErrInvalidPortsRangeString, port)
+	}
+
+	startPort, err := strconv.Atoi(ports[0])
+	if err != nil {
+		return 0, p2p.ErrInvalidStartingPortValue
+	}
+
+	endPort, err := strconv.Atoi(ports[1])
```

Comment:
This does not should be checked for negative value?

Reply:
it is checked against startPort which should be greater than minRangePortValue